packages/actor-extract-links-extract-tree/lib/ActorExtractLinksTree.ts

```diff
@@ -41,15 +51,26 @@ export class ActorExtractLinksTree extends ActorExtractLinks {
       metadata.on('data', (quad: RDF.Quad) =>
         this.getTreeQuadsRawRelations(quad,
           currentNodeUrl,
-          pageRelationNodes,
-          nodeLinks));
+          relationNodeSubject,
+          nodeLinks,
+          effectiveTreeDocumentSubject));
 
       // Resolve to discovered links
       metadata.on('end', () => {
-        // Validate if the node forward have the current node as implicit subject
-        for (const [ nodeValue, link ] of nodeLinks) {
-          if (pageRelationNodes.has(nodeValue)) {
-            links.push({ url: link });
+        // If we are not in the loose mode then the subject of the page is the URL
+        if (effectiveTreeDocumentSubject.size === 0) {
+          effectiveTreeDocumentSubject.add(currentNodeUrl);
+        }
+        // If there are multiple subjects for the node, we consider that the page is inconsistent
+        // and rejects all the relations. It has to be noted that we don't support owl reasoning.
```

Comment:
Does it make sense to consider all of them in that case, instead of rejecting?
If not, then we could also consider emitting a warning to the logger.